### PR TITLE
Integration spec to views

### PR DESCRIPTION
**Pull request by andersonlebon**
In this PR, we did the following:

- [ ] Use Capybara to write integration tests for each view.
- [ ] Login page:
- I can see the username and password inputs and the "Submit" button.
- When I click the submit button without filling in the username and the password, I get a detailed error.
- When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
- When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
- [ ] User index page:
- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.
- [ ] user show page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.
- [ ] User post index page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.
- [ ] Post.show page:
- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commenter.
- I can see the comment each commenter left.

### CODE REVIEW

File: spec/features/log_in_feature_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        fill_in 'user_password', with: 'password'
+      end
+      click_button 'Sign in'
+      expect(page).to have_content 'Invalid Email or password'
+    end
+
+    scenario 'with correct email and password' do
+      within 'form' do
+        fill_in 'user_email', with: ''
+        fill_in 'user_password', with: ''
```
Comment (Davidosky007) [inline spec/features/log_in_feature_spec.rb]: * Kindly add the following tests as it is required `Tests that redirects to the root page when a correct username and password is entered.`
* Please fix line `29` scenario `with correct email and password` as the test should yield a positive and not a negative result

File: spec/features/show_user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    scenario 'I can see the number of posts each user has written' do
+      within 'form' do
+        fill_in 'user_email', with: 'user@example.com'
+        fill_in 'user_password', with: 'password'
+      end
+      click_button 'Sign in'
+      expect(page).to have_content 'Invalid Email or password'
+    end
```
Comment (Davidosky007) [inline spec/features/show_user_index_spec.rb]: * Kindly fix these tests as the result doesn't match the scenario for example `context` : `i can see the profile picture for each user ` == `Invalid Email or password` please add these test and the expected results 
 * Tests that asserts you can see the username of all other users.
* Tests that asserts you can see the profile picture for each user.
* Tests that asserts you can see the number of posts each user has written.
* Tests that redirects to the user's show page when you click on a user. [ checkout this Capybara cheat sheet ](https://gist.github.com/zhengjia/428105)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @andersonlebon 👋, 

Good job so far! 💪 
* No linter errors ✔️ 
* Descriptive commit messages ✔️ 
* Well written PR ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-jumpin-gif


## Required Changes ♻️
* Kindly write all tests to give the expected result as the test are giving results different from their scenario [Here is a capybara cheat sheet to help you ](https://gist.github.com/zhengjia/428105)
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉**
Hello @andersonlebon 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**